src/include/libpmemfile-core-stubs.h

```diff
@@ -0,0 +1,130 @@
+ */
+
+/*
+ * libpmemfile-core-stubs.h -- definitions of not yet implemented
+ * libpmemfile-core entry points. Do not use these. All the routines
+ * just set errno to ENUTSUP.
+ * This is header file, and the symbols exported are used while designing
+ * the interface of the library.
+ * Everything here is subject to change at any time.
+ *
+ * If/when some pmemfile functionality is implemented, the corresponding
+ * header declarations should be moved to the libpmemfile-core.h header file.
+ *
+ * This file is expected to be removed eventually.
+ */
+#ifndef LIBPMEMFILE_CORE_H
+#error Never include this header file directly
+#endif
+
+#ifndef LIBPMEMFILE_CORE_STUBS_H
+#define LIBPMEMFILE_CORE_STUBS_H
+
+struct linux_dirent;
+struct linux_dirent64;
+
+int pmemfile_chdir(PMEMfilepool *, const char *path);
+int pmemfile_fchdir(PMEMfilepool *, PMEMfile *);
+int pmemfile_stat(PMEMfilepool *, const char *path, struct stat *buf);
+int pmemfile_lstat(PMEMfilepool *, const char *path, struct stat *buf);
+int pmemfile_fstat(PMEMfilepool *, PMEMfile *file, struct stat *buf);
+int pmemfile_readlink(PMEMfilepool *, const char *path,
+			char *buf, size_t buf_len);
+char *pmemfile_getcwd(PMEMfilepool *, char *buf, size_t buf_len);
+ssize_t pmemfile_pread(PMEMfilepool *, PMEMfile *file,
+			char *buf, size_t count, off_t pos);
+ssize_t pmemfile_pwrite(PMEMfilepool *, PMEMfile *file,
+			const char *buf, size_t count, off_t pos);
+int pmemfile_access(PMEMfilepool *, const char *path, mode_t mode);
+int pmemfile_sync(PMEMfilepool *, PMEMfile *);
+int pmemfile_fsync(PMEMfilepool *, PMEMfile *);
+int pmemfile_fdatasync(PMEMfilepool *, PMEMfile *);
+int pmemfile_rename(PMEMfilepool *, const char *old_path, const char *new_path);
+int pmemfile_renameat(PMEMfilepool *, PMEMfile *old_at, const char *old_path,
+				PMEMfile *new_at, const char *new_path);
+int pmemfile_openat(PMEMfilepool *, PMEMfile *at,
+		const char *path, int oflag, mode_t mode);
+int pmemfile_flock(PMEMfilepool *, PMEMfile *file, int operation);
+int pmemfile_truncate(PMEMfilepool *, const char *path, off_t length);
+int pmemfile_ftruncate(PMEMfilepool *, PMEMfile *file, off_t length);
+int pmemfile_getdents(PMEMfilepool *, PMEMfile *file,
+			struct linux_dirent *dirp, unsigned count);
+int pmemfile_getdents64(PMEMfilepool *, PMEMfile *file,
+			struct linux_dirent64 *dirp, unsigned count);
+int pmemfile_mkdir(PMEMfilepool *, const char *path, mode_t mode);
+int pmemfile_mkdirat(PMEMfilepool *, PMEMfile *file,
+			const char *path, mode_t mode);
+int pmemfile_rmdir(PMEMfilepool *, const char *path);
+int pmemfile_symlink(PMEMfilepool *, const char *path1, const char *path2);
+int pmemfile_symlinkat(PMEMfilepool *, const char *path1,
+				PMEMfile *at, const char *path2);
+
+/*
+ * Do we need pmemfile_readahead ? It would be a NOP, wouldn't it?
+ * ssize_t pmemfile_readahead(PMEMfilepool *, PMEMfile *file,
+ *				off64_t offset, size_t count);
+ */
+
+/*
+ * What about chmod, chown, etc... ?
```

Comment:
I honestly think that chown should return ENOTSUP, unless it is really needed by some application. Since we can't effectively alter any ownership, it wouldn't be a good idea to use apps, that rely on it.